### DISCUSSION TITLE
How is data loaded?

### DISCUSSION THREAD

**Discussion by mavaylon1** [mrdbourke/pytorch-deep-learning#588] [Q&A]
Most of the tutorials online show data being wrapped with a DataLoader. In the "01 Fundamentals" section, the data is stored in a tensor. How is this data iterated and loaded into the model? From my understanding the model takes one number at a time from the train X and y when training since we define only one weight and bias.

**Comment by mrdbourke** [comment]
Hi @mavaylon1 , 

Good question!

For larger datasets, most of the time you'll be using a PyTorch DataLoader (so data be turned into batches, we talk about these later in the course).

But for smaller datasets (like in section [01. PyTorch Workflow Fundamentals](https://www.learnpytorch.io/01_pytorch_workflow/#pytorch-training-loop)), the dataset is so small we don't need to turn it into batches.

This way we can pass the entire dataset `X_train` and `X_test` to the model in one go.

**Reply by mavaylon1** [reply]
What do you mean it takes the entire dataset? With each forward pass it takes only one datapoint at a time, iterating through the set right? Essentially a batch size of 1. 

Or are we feeding the entire set as our input at once and not iterating at all?
